packages/design/src/SlideTabs/SlideTabs.test.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import SlideTabs from './SlideTabs';
+import { render, fireEvent } from 'design/utils/testing';
+
+describe('design/SlideTabs', () => {
+  it('renders the supplied number of tabs(3)', () => {
+    const { container } = render(
+      <SlideTabs tabs={['aws', 'automatically', 'manually']} />
+    );
+    expect(container.getElementsByTagName('label')).toHaveLength(3);
+  });
+
+  it('renders the supplied number of tabs(5)', () => {
+    const { container } = render(
+      <SlideTabs
+        tabs={['aws', 'automatically', 'manually', 'apple', 'purple']}
+      />
+    );
+    expect(container.getElementsByTagName('label')).toHaveLength(5);
+  });
+
+  it('respects a custom form name', () => {
+    const { container } = render(
+      <SlideTabs name="pineapple" tabs={['aws', 'automatically', 'manually']} />
+    );
+    expect(container.querySelectorAll('input[name=pineapple]')).toHaveLength(3);
+  });
+
+  it('calls the onChange handler when the tab is changed', () => {
+    const cb = jest.fn();
+    const { container } = render(
+      <SlideTabs onChange={cb} tabs={['aws', 'automatically', 'manually']} />
+    );
+    fireEvent.click(container.querySelector('label[for=slide-tab-manually]'));
+    expect(cb.mock.calls).toHaveLength(2);
+    expect(cb.mock.calls).toEqual([[0], [2]]);
```

Comment:
wasnt sure i was reading it right, does this mean `onChange` is getting called twice for one click?

Reply:
Good question, I'll add a comment to this effect.

But no, what's happening is that when the component initializes it's setting the tab to index `0` then when you click it's setting it to index `2`. This was intentional because this tab control will be the source of truth for the Slider so it needs to know what to display first as you could, in theory, have it start at a different index.